packages/plugin-js/src/index.ts

```diff
@@ -102,7 +114,7 @@ function defaultTransform(token: ParsedToken, mode?: string): (typeof token)['$v
   return {...(token.$value as typeof modeVal), ...modeVal};
 }
 
-function setToken(obj: Record<string, any>, id: string, value: any, nest = false) {
+function setToken<T extends Record<string, any>>(obj: T, id: keyof T, value: any, nest = false): T {
```

Comment:
This type change is irrelevant; just passes inference from the original object through better. But has no impact on the current code.